sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -885,20 +884,6 @@ private[hive] class HiveClientImpl(
       // Since HIVE-18238(Hive 3.0.0), the Driver.close function's return type changed
       // and the CommandProcessorFactory.clean function removed.
       driver.getClass.getMethod("close").invoke(driver)
-      if (version != hive.v3_0 && version != hive.v3_1 && version != hive.v4_0) {
```

Comment:
This would break backward compatibility when spark tries to connect with older version of Hive right?

Reply:
I don't think it is a connection problem. There will be a problem when 2.x Hive jars are provided at runtime. It is not clear to me if it is still necessary to support such option granted that 2.x is EOL.

Reply:
2.x is ancient and should no longer be supported

Reply:
Agreed. 
Hive 2.x and 3.x is EoL. We should move completely to hive 4+.

Thanks!